### PR TITLE
Added MalShare searching

### PR DESCRIPTION
**Pull request by silascutler**
Added search functionality to link in MalShare.com

### CODE REVIEW

File: static/script.js [hunk 1]
```diff
@@ -9,6 +9,20 @@ function hash(path, handler) {
     });
 }
 
+/* --- VirusTotal --- */
```
Comment (davidt99) [inline static/script.js]: rename to MalShare

File: config.py [hunk 2]
```diff
@@ -1,2 +1,3 @@
 VT_APIKEY = '<KEY>'
-OTX_APIKEY = '<KEY>' # optional
\ No newline at end of file
+OTX_APIKEY = '<KEY>' # optional
+MALSHARE_APIKEY = '<KEY>' #OPTIONAL
```
Comment (davidt99) [inline config.py]: can you change the comment to `# optional`

File: linux_explorer.py [hunk 3]
```diff
@@ -125,6 +125,16 @@ def find_by_type(type):
 
     return jsonify(otx.get_indicator_details_full(find_by_type(type), indicator)), 200
 
+@app.route('/malshare/<string:type>/<string:indicator>')
+def malshare_report(type, indicator):
+    ''' MalShare repository.
+        Supported indicator types: "domain", "hostname", "URL", "FileHash-MD5", "FileHash-SHA1", "FileHash-SHA256", '''
+
+    response = requests.get('https://malshare.com/api.php?api_key=%s&action=details&hash=%s' % ( config.MALSHARE_APIKEY, indicator ), headers={'Accept-Encoding': 'gzip, deflate', 'User-Agent': 'gzip,  Linux Expl0rer'})
```
Comment (davidt99) [inline linux_explorer.py]: can you handle no api key similar to [otx](https://github.com/intezer/linux-explorer/pull/1/files#diff-24c478777ae4b5954679a8c3125a85f1R121)?

File: templates/processes.html [hunk 4]
```diff
@@ -12,9 +12,18 @@
                 $('#sha256_' + d.pid).html(sha256);
 
                 $('#vt_' + d.pid).html("searching...");
+                $('#malshare_' + d.pid).html("searching...");
                 $('#otx_' + d.pid).html("searching...");
 
                 /* --- Search in VirusTotal and OTX --- */
```
Comment (davidt99) [inline templates/processes.html]: Add MalShare to the comment

File: static/script.js [hunk 5]
```diff
@@ -9,6 +9,20 @@ function hash(path, handler) {
...
+function MalShare_search(sha256, handler, interval=5000) {
+    $.get("/malshare/sha256/" + sha256, function(data, status) {
+        if (status == "success") {
+            handler(data);
+        } else {
+            setTimeout(function () {
+                MalShare_search(sha256, handler, interval); // retry until success
```
Comment (davidt99) [inline static/script.js]: there is a reason to retry? if it failed, i think it's better to just write an error instead of retrying

**Comment by davidt99** [comment]
@silascutler ping. If you authorize it in github, I can make the changes in your repository and merge it.

**Comment by silascutler** [comment]
Added.  Thanks

**Comment by davidt99** [comment]
@silascutler I changed what I wanted, Thank you for the PR!

**Pull request merged**